src/pudl/extract/csv.py

```diff
@@ -1,72 +1,52 @@
 """Extractor for CSV data."""
-from csv import DictReader
-from importlib import resources
-from zipfile import ZipFile
-
 import pandas as pd
 
 import pudl.logging_helpers
+from pudl.extract.extractor import GenericExtractor, PartitionSelection
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-def open_csv_resource(dataset: str, base_filename: str) -> DictReader:
-    """Open the given resource file as :class:`csv.DictReader`.
-
-    Args:
-        dataset: used to load metadata from package_data/{dataset} subdirectory.
-        base_filename: the name of the file in the subdirectory to open.
-    """
-    csv_path = resources.files(f"pudl.package_data.{dataset}") / base_filename
-    return DictReader(csv_path.open())
-
-
-def get_table_file_map(dataset: str) -> dict[str, str]:
-    """Return a dictionary of table names and filenames for the dataset.
-
-    Args:
-        dataset: used to load metadata from package_data/{dataset} subdirectory.
-    """
-    return {
-        row["table"]: row["filename"]
-        for row in open_csv_resource(dataset, "table_file_map.csv")
-    }
-
-
-class CsvExtractor:
-    """Generalized class for extracting dataframes from CSV files.
+class CsvExtractor(GenericExtractor):
+    """Class for extracting dataframes from CSV files.
 
     The extraction logic is invoked by calling extract() method of this class.
     """
 
-    def __init__(self, zipfile: ZipFile, table_file_map: dict[str, str]):
-        """Create a new instance of CsvExtractor.
-
-        This can be used for retrieving data from CSV files.
+    def source_filename(self, page: str, **partition: PartitionSelection) -> str:
```

Comment:
Type hints for **kwargs are not intuitive to me (we're actually working with a `dict` in `**partitions`) but I found defining `PartitionSelection` useful for me to wrap my head around the intention and use of the variables here. Happy to remove it if it isn't helpful.